client/components/FormattingBar/buttons.ts

```diff
@@ -182,6 +187,57 @@ export const code: FormattingBarButtonData = {
 	command: codeToggle,
 };
 
+const isInCodeBlock = (editorChangeObject: EditorChangeObject): boolean => {
+	if (!editorChangeObject.view) return false;
+	const state = editorChangeObject.view.state;
+	const { $anchor } = state.selection;
+	for (let d = $anchor.depth; d > 0; d--)
+		if ($anchor.node(d).type.name === 'code_block') return true;
+	return false;
```

Comment:
You could use the previously defined `isInCodeBlock` function from `code.ts` to de-dupe some of this code.